tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTSecurityCRLTest.java

```diff
@@ -0,0 +1,357 @@
+      return new File(new File(protectionDomain.getCodeSource().getLocation().getPath()), "../..").getCanonicalFile();
+   }
+
+   @Test(expected = SSLException.class)
+   public void crlRevokedTest() throws Exception {
+
+      ActiveMQServer server1 = initServer();
+      BlockingConnection connection1 = null;
+      try {
+         server1.start();
+
+         while (!server1.isStarted()) {
+            Thread.sleep(50);
+         }
+
+         String basedir = basedir().getPath() + "/src/test/resources/mqttCrl/client0/";
+         connection1 = retrieveMQTTConnection("ssl://localhost:1883", basedir + "truststore.jks", "changeit", basedir + "client_revoked.jks", "changeit");
+
+         // Subscribe to topics
+         Topic[] topics = {new Topic("test/+/some/#", QoS.AT_MOST_ONCE)};
+         connection1.subscribe(topics);
+
+         // Publish Messages
+         String payload1 = "This is message 1";
+
+         connection1.publish("test/1/some/la", payload1.getBytes(), QoS.AT_LEAST_ONCE, false);
+
+         Message message1 = connection1.receive(5, TimeUnit.SECONDS);
+
+         assertEquals(payload1, new String(message1.getPayload()));
+
+      } finally {
+         if (connection1 != null) {
+            connection1.disconnect();
+         }
+         if (server1.isStarted()) {
+            server1.stop();
+         }
+      }
+
+   }
+
+   @Test
+   public void crlNotRevokedTest() throws Exception {
+
+      ActiveMQServer server1 = initServer();
+      BlockingConnection connection1 = null;
+      try {
+         server1.start();
+
+         while (!server1.isStarted()) {
+            Thread.sleep(50);
+         }
+
+         String basedir = basedir().getPath() + "/src/test/resources/mqttCrl/client1/";
+         connection1 = retrieveMQTTConnection("ssl://localhost:1883", basedir + "truststore.jks", "changeit", basedir + "client_not_revoked.jks", "changeit");
+
+         // Subscribe to topics
+         Topic[] topics = {new Topic("test/+/some/#", QoS.AT_MOST_ONCE)};
+         connection1.subscribe(topics);
+
+         // Publish Messages
+         String payload1 = "This is message 1";
+
+         connection1.publish("test/1/some/la", payload1.getBytes(), QoS.AT_LEAST_ONCE, false);
+
+         Message message1 = connection1.receive(5, TimeUnit.SECONDS);
+
+         assertEquals(payload1, new String(message1.getPayload()));
+
+      } finally {
+         if (connection1 != null) {
+            connection1.disconnect();
+         }
+         if (server1.isStarted()) {
+            server1.stop();
+         }
+      }
+
+   }
+
+
+   private ActiveMQServer initServer() throws Exception {
+      Configuration configuration = createConfiguration("broker");
+      ActiveMQServer server = createServer(true, configuration);
+      configureBrokerSecurity(server);
+      return server;
+   }
+
+   protected Configuration createConfiguration(String name) throws Exception {
+      FileConfiguration fc = new FileConfiguration();
```

Comment:
There are already lots of methods in the test-suite to create a config for a test.  Take a look at org.apache.activemq.artemis.tests.util.ActiveMQTestBase#createDefaultNettyConfig for example.